sdk/core/core-lro/src/lroEngine/lroEngine.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Poller } from "../poller";
+import { PollOperationState } from "../pollOperation";
+import { LongRunningOperation, LroEngineOptions, ResumablePollOperationState } from "./models";
+import { GenericPollOperation } from "./operation";
+
+/**
+ * The LRO Engine, a class that performs polling.
+ */
+export class LroEngine<TResult, TState extends PollOperationState<TResult>> extends Poller<
+  TState,
+  TResult
+> {
+  private intervalInMs: number;
+
+  constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions) {
+    const { intervalInMs = 2000, resumeFrom } = options || {};
+    function deserializeState(
```

Comment:
having a function declared inside the constructor body feels a bit strange. Can we hoist this out of the class into module scope?

Reply:
Done.